### ISSUE TITLE
Find a way to estimate the test coverage of the project.

### ISSUE THREAD

**Issue by iocanel** [dekorateio/dekorate#510]
Estimating the coverage is `dekorate` is a bit tricky, because the actual test code doesn't invoke dekorate itself and just verifies the result. So, dekorate runs during the `compilation` of the module and  then test runs during `testing` phase.

We need to figure a way to overcome this issue and get the actual data.

**Comment by iocanel** [comment]
Status: Tried jacoco maven plugin but it completely breaks the build with weird classcasting errors.

**Comment by iocanel** [comment]
**Status update**: Tried using `jacoco`. It seems that it performs some sort of instrumentation that is causing a series of `ClassCastException` and thus preventing the project from building.

**Comment by cmoulliard** [comment]
Did you chated about that with Quarkus or MW projects to figure out if there is a recommended project doing well this job ? @iocanel

**Comment by iocanel** [comment]
@cmoulliard: No, we are mostly black box testing and I am not sure if traditional strategies apply.

**Comment by iocanel** [comment]
*Status*: The issue is stale as I had no time to work on it.
Maybe we should move it back to backlog/icebox?

**Comment by iocanel** [comment]
@Jose I assigned this to you in case you have any ideas on what we could do for this. If you don't anything to suggest feel free to unassign as it's not a priority.

**Comment by Sgitario** [comment]
I've been trying to find a solution for this but taking into account the nature of Dekorate which generates resources that are then installed on a cluster and then indirectly used by the tests, we can't calculate or estimate what is covered and what's not. 

We could certainly configure jacoco to record the decorators and the code that is invoked/used at Maven build, but still, we would not know if the decorator output is actually used/covered or not.
